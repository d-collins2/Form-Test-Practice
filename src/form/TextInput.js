import React from 'react';

const TextInput = ({ value, handleChange, formError, name, placeholder, controlId }) => {
    function errorClass(error) {
        if(error){
            return( error.length === 0 ? '' : 'has-error');
        }
    }
    return (
        <div className={`form-group ${errorClass(formError)}`} id={controlId}>
            <input
                required
                className="form-control"
                value={value}
                onChange={handleChange}
                type="text"
                name={name}
                placeholder={placeholder}/>
            <span className="help-block text-red">{formError}</span>
            <div className="text-right">
                <span><small className="text-muted">Required</small></span>
            </div>
        </div>
    )
}

export default TextInput;
