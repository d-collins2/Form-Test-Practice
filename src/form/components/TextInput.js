import React from 'react';
import { Form } from 'react-bootstrap'

const TextInput = ({ value, handleChange, formError, name, placeholder, controlId }) => {
    function errorClass(error) {
        if(error){
            return( error.length === 0 ? '' : 'has-error');
        }
    }
    return (
        <Form.Group className={errorClass(formError)} controlid={controlId}>
            <Form.Control
                required
                value={value}
                onChange={handleChange}
                type="text"
                name={name}
                placeholder={placeholder}/>
            <Form.Text className="help-block text-red">{formError}</Form.Text>
            <div className="text-right">
                <Form.Text className="text-muted">Required</Form.Text>
            </div>
        </Form.Group>
    )
}

export default TextInput;
