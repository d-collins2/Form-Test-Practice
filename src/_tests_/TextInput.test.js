import React from 'react';
import renderer from 'react-test-renderer';
import { Form } from 'react-bootstrap'
import { shallow } from 'enzyme';
import TextInput from '../form/TextInput';

describe('<TextInput />', function() {

    it('Should render correctly', function(){
        const component = renderer.create(<TextInput />);
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('Should call changeHandler onChange on input', function() {
        const mockHandler = jest.fn();
        const component = shallow(<TextInput handleChange={mockHandler}/>);
        const input = component.find(Form.Control);
        input.simulate('change');
        expect(mockHandler.mock.calls.length).toBe(1);
    })
})
