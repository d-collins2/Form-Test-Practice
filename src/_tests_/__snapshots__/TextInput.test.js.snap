// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextInput /> Should render correctly 1`] = `
<div
  className="has-error form-group"
  controlid="Id"
>
  <input
    className="form-control"
    name="Name"
    onChange={[Function]}
    placeholder="Placeholder"
    required={true}
    type="text"
    value="Label"
  />
  <small
    className="help-block text-red form-text"
  >
    Error
  </small>
  <div
    className="text-right"
  >
    <small
      className="text-muted form-text"
    >
      Required
    </small>
  </div>
</div>
`;
